File: Leetcode/two_pointers/283_MoveZeroes/ArrayPartition/Solution.java
public class Solution {
    /**
     * @param nums: The integer array you should partition
     * @param k: An integer
     * @return: The index after partition
     */
    public int partitionArray(int[] nums, int k) {
        // write your code here
        int lastSmallerIndex = 0, temp;
        for (int i = 0; i<nums.length; i++){
            if (nums[i]<k){
                temp = nums[lastSmallerIndex];
                nums[lastSmallerIndex] = nums[i];
                nums[i] = temp;
                lastSmallerIndex++;
            }
        }
        return lastSmallerIndex;
    }
}

File: Leetcode/two_pointers/283_MoveZeroes/ArrayPartition/Solution.py
class Solution:
    """
    @param nums: The integer array you should partition
    @param k: An integer
    @return: The index after partition
    """
    def partitionArray(self, nums, k):
        # write your code here
        temp = list(filter(lambda x: x<k, nums))
        nums = temp + list(filter(lambda x: x>=k, nums))
        return len(temp)
